examples/planning/python/planner_client.py

```diff
@@ -0,0 +1,232 @@
+import grpc
+import sys
+import yaml
+import os
+import click
+import os.path
+from typing import Mapping, List, Any
+import numpy as np
+
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+DATA_DIR = os.path.join(os.path.dirname(ROOT_DIR), "data")
+BUILDFILES_DIR = os.path.join(ROOT_DIR, "build")
+if BUILDFILES_DIR not in sys.path:
+    sys.path.append(BUILDFILES_DIR)
+
+from build.basic_types_pb2 import PlanContextId, ProblemDef, SystemConf, Conf, SysPoly
+from build.planner_pb2 import (
+    StartPlanRequest,
+    StartPlanResponse,
+    RetrievePlanRequest,
+    RetrievePlanResponse,
+)
+from build.planner_pb2_grpc import MotionPlannerStub
+
+K = 4
+
+
+class PiecewiseCubicPolynomial:
+    """Custom implementation for a piecewise polynomial trajectory."""
+
+    def __init__(self, breaks, P_arr: np.ndarray):
+        self.degree = K
+        self.breaks = np.array(breaks)
+        self.start, self.end = self.breaks[0], self.breaks[-1]
+        if P_arr.shape[-1] != self.degree:
+            raise ValueError(
+                f"Polynomial array of shape: {P_arr.shape} is not permitted for a cubic piecewise polynomial"
+            )
+        self.P_arr = P_arr
+
+    def rows(self):
+        return self.P_arr.shape[1]
+
+    def cols(self):
+        return self.P_arr.shape[2]
+
+    def value(self, t: float) -> np.ndarray:
+        """Compute the value of the piecewise polynomial at time t.
+
+        Args:
+            t (float): time
+
+        Returns:
+            np.ndarray: joint positions at time t
+        """
+        if t < self.start or t > self.end:
+            raise ValueError(
+                f"Polynomial is undefined for t={t} outside [{self.start}, {self.end}]!"
+            )
+        t_vec = np.array([[1, t, t**2, t**3]])
+        if t == self.start:
+            P_idx = 0
+        elif t == self.end:
+            P_idx = self.P_arr.shape[0] - 1
+        else:
+            P_idx = np.searchsorted(self.breaks[:-1], t) - 1
```

Comment:
docs [here](https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html). Essentially a binary search in order to grab the interval on which the sample time `t` should be evaluated.